CONTRIBUTING.md

```diff
@@ -1,9 +1,85 @@
 # Contributing Guidelines
 
-Here we are going to tell you how to contribute. This is a work in progress
-document, and will improve in the future.
+👍🎉 First off, thanks for taking the time to contribute! 🎉👍
+
+The following is a set of guidelines for contributing to **Privacy evaluator for Machine Learning models**, which is an open-source project hosted on GitHub. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+You can also find [README](https://github.com/privML/privacy-evaluator) and [CODE OF CONDUCT](https://github.com/privML/privacy-evaluator).
+
+# How Can I Contribute
+
+## Issues
+In order to achieve an efficient workflow, there should be an easily accessible resource to get information on the current state of development. We use [Projects table](https://github.com/orgs/privML/projects/1) for that. That way, a developer can quickly select an open issue that is ready for implementation and stakeholders can get an overview of the progress. Each time you want to solve a new problem, create an Issue, give a short description of the problem, its acceptance criteria, assign people, tag if needed.
+
+## Branches
+
+### Special branches
+
+- `main`: the current state of development
```

Comment:
Hmm do we maybe need a branch for each team or/and sprint? Does that even matter in the guidelines because they're mostly for outsiders right?

Reply:
I mean, it is for both outsiders and us. We will be actually working on this, don't expect outside rush on our code ;)
But I don't think two branches are needed for two teams, since we work on the same project which needs to be up-to-date, and our sprints are alternating.

Reply:
Haha yes me neither. 
Hmm I still think that branches per sprint could make sense, then the main branch would always only contain working increments. For now it does not matter anyway, because we don't have any increments yet.

Reply:
alternatively we could also do an `incr-release` branch and let the main branch be the work in progress branch. I'd like it more the other way around but in the end it does not really matter.

Reply:
> because they're mostly for outsiders right?

I would try to have no _outsider_ vs _insider_ type of division, to be inclusive of the outside community, and have all the communication and rules the same for everybody as much as possible.

As for branching, I'd recommend having `main` as the branch where all contributions are made to, and have it always pass the tests and it be a functional branch. Then you can have a different branch with appropriate tags for each release.

As for contributions, ideally each contribution is something which is in a single PR and contains all tests and implementation for a single feature.

issues and projects should be organized in a way that they can fit reasonably in a single PR, but you can split implementation from documentation if you want, although I'd discourage that. However, major documentation contributions are their own PR.

The concept of a sprint shouldn't leak into your development, and your main branch and CI/CD should be independent of those cycles. This is partly due to the reason that sprints are for _you_ and not _outsiders_. If there are contributions where two or more people work on it, one can create a PR, and the other person can submit a PR into that other person's PR branch and they can work on it together. For instance, let say here we have:

- `privML/privacy-evaluator:main`: the `main` branch of the _main_ repository.
- `user1/privacy-evaluator:feature1`: the _fork_ under `user1` who is developing `feature1`. The PR would be `user1/privacy-evaluator:feature1` -> `privML/privacy-evaluator:main`.
- `user2/privacy-evaluator:feature1-complement`: If `user2` wants to add things to the implementation which is under `user1/privacy-evaluator:feature1`, they can have their own for (from the _main_ repo), but introduce a third `remote` which refers to `user1/privacy-evaluator` to sync with that one if needed. Then they would create a PR in the form of `user2/privacy-evaluator:feature1-complement` -> `user1/privacy-evaluator:feature1`. Once this PR is merged into that branch, it'll be also visible in the `user1/privacy-evaluator:feature1` -> `privML/privacy-evaluator:main` PR.

I know it sounds complicated at the beginning, and happy to clarify on points which may need more clarification.